### PR TITLE
ci: Add pre-release to release workflow

### PR DESCRIPTION
**Pull request by seanpoulter**
### Proposed Changes

- Release when we publish `latest`
- Pre-release when we publish `next` (or not `latest`)
- Document the release process
﻿
### Notes

- I didn't quote the workflow inputs so we can still add `--dry-run`.

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -76,3 +76,21 @@ Check the issues or [raise a new one](https://github.com/webdriverio-community/w
...
+
+Project maintainers can publish a release or pre-release of the npm package by manually running the [`Manual NPM Publish`](https://github.com/webdriverio-community/wdio-electron-service/actions/workflows/release.yml) GitHub Workflow. They will choose the release type to trigger a `major` , `minor`, or `patch` release following [Semantic Versioning](https://semver.org/).
+
+To publish a release, run the Workflow with default **Branch** `main` and **NPM Tag** `latest` and the appropriate **Release Type**. This will:
+
+* Create a Git tag
+* Create a GitHub Release
+* Publish to npm
+
+To publish a pre-release, also referred to as a test release, run the Workflow with the **NPM Tag** `next`. This will:
```
Comment (christian-bromann) [inline docs/development.md:90]: Can we add a note which version bump to pick when making a pre-release?

**Review by christian-bromann — APPROVED** [review]
One comment, otherwise LGTM 👍 
Will let @goosewobbler review as well and have him merge.

Thanks for adding docs to this!

File: docs/development.md (hunk 1 above)
Comment (seanpoulter) [reply docs/development.md:90]: Ah, I missed that it'd be a `patch`. :+1:

**Comment by goosewobbler** [comment]
This looks good to me, I just had a query about the syntax for the release-it commands not matching what was previously there.  Any reason for closing -- are you recreating it?

**Comment by seanpoulter** [comment]
> Any reason for closing -- are you recreating it?

:facepalm:

--

> I just had a query about the syntax for the release-it commands not matching what was previously there

Yea, here's where a quick call would help. Let's walk through it. We used to have:

```
run: pnpm run release:ci --npm.tag=${{github.event.inputs.distTag}} -- ${{github.event.inputs.releaseType}}
```

Based on [the docs](https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputs), I simplified `github.event.inputs.` as `inputs.`:

> The workflow will also receive the inputs in the `github.event.inputs` context. The information in the `inputs` context and `github.event.inputs` context is identical except that the `inputs` context preserves Boolean values as Booleans instead of converting them to strings. The choice type resolves to a string and is a single selectable option.

We're not using Boolean's so that leads to a bit less noise.

--

I changed the argument order to look a bit more like the [release-it docs](https://github.com/release-it/release-it/blob/main/docs/pre-releases.md):

```
release-it major --preRelease=beta
```
so
```diff
run: pnpm run release:ci --npm.tag=${{github.event.inputs.distTag}} -- ${{github.event.inputs.releaseType}}
run: pnpm run release:ci ${{inputs.releaseType}} --npm.tag=${{inputs.distTag}}
```
--

Then I split the release and pre-release. The release should look like line above with "no changes":
```
run: pnpm run release:ci ${{inputs.releaseType}} --npm.tag=${{inputs.distTag}}
```
And the pre-release should use `--preRelease` instead of `--npm.tag` because `--preRelease` is shorthand for:
```
--preReleaseId=${{inputs.distTag}} --npm.tag=${{inputs.distTag}} --github.preRelease
```
That looks like
```
run: pnpm run release:ci ${{inputs.releaseType}} --preRelease=${{inputs.distTag}}
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -48,8 +49,17 @@ jobs:
...
+        if: ${{ inputs.distTag != 'latest' }}
+        run: pnpm run release:ci ${{inputs.releaseType}} --preRelease=${{inputs.distTag}}
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
       - name: Release
-        run: pnpm run release:ci --npm.tag=${{github.event.inputs.distTag}} -- ${{github.event.inputs.releaseType}}
+        if: ${{ inputs.distTag == 'latest' }}
+        run: pnpm run release:ci ${{inputs.releaseType}} --npm.tag=${{inputs.distTag}}
```
Comment (goosewobbler) [inline .github/workflows/release.yml:62]: Just checking if this is functionally the same as the line it is replacing or if we do need the ` -- ` before the releaseType.

Comment (goosewobbler) [reply .github/workflows/release.yml:62]: Explained in the comment above.

**Comment by goosewobbler** [comment]
Nice, I'll merge this then as soon as the tests pass

**Pull request merged**